### ISSUE TITLE
Right-click in House Preview corrupts Selected Model

### ISSUE THREAD

**Issue by leighfiddes** [smeighan/xLights#1964]
NOTE: IF YOU DO NOT ATTACH A SEQUENCE THAT DEMONSTRATES YOUR PROBLEM THEN THERE IS A HIGH PROBABILITY YOUR ISSUE WILL JUST BE CLOSED AND NOT INVESTIGATED.

**Describe the bug**
On the Layout page, right-clicking the mouse in the House Preview will corrupt the selected model is any.

**To Reproduce**
1. Go to layout tab.
2. Select a model on the house preview window.
3. Right-click anywhere in the house preview to bring up the menu
4. Observer that the selected model has been expanded and rotated.

**Expected behavior**
The right-click action should not corrupt the model.

**Screenshots**
Before and After screenshots:
![Screenshot_20200308_before](https://user-images.githubusercontent.com/59780576/76156029-8c715a80-6148-11ea-8bbf-05d533f51ac0.png)

![Screenshot_20200308_after](https://user-images.githubusercontent.com/59780576/76156025-68ae1480-6148-11ea-8457-1989da2e1237.png)

**Versions (please complete the following information):**
 - Kubuntu Linux 18.04 LTS
 - xLights version 2020.4

**Additional context**
Issue is observed on all xLights 2020.x
Reset or Undo does not restore the model.

**Attachments**
- If associated with a crash please attach/link to a dropbox containing the crash log that was produced
- If it crashed but no crash log was produced please use Tools Menu/Package Logs and attach/link to a dropbox containing the resulting zip file
- If the error relates to rendering an effect or a crash while sequencing please use Tools Menu/Package Sequence and attach/link to a dropbox containing the resulting zip file

If you would prefer not to post your sequences publically please say so and be ready to PM them to one of the developers when asked.

**Comment by keithsw1111** [comment]
I cant reproduce this on windows.

**Comment by leighfiddes** [comment]
My Analysis:
I am running Kubuntu 18.04 LTS on a desktop PC (not a laptop) with xLights 2020.x built with wxWidgets 3.1.3 which is built with GTK3. Am I the only user running this scenario?

When I right-click inside the House Preview window, to get the pop-up menu, as well as the menu being shown, the following two events are generated in this order:
1) wxEVT_GESTURE_ZOOM with Zoom Factor = 1, Start = false, End = true.
2) wxEvt_GESTURE_ROTATE with Angle = 0, Start = false, End = true.

I have not been able to identify WHY these two events are being generated. It appears to be an issue in wxGTK when no touchpad is available perhaps.

Looking at the code in LayoutPanel::OnPreviewZoomGesture and LayoutPanel::OnPreviewRotateGesture it is easy to see that, due to these Gesture events not having corresponding Start=true events, then m_last_mouse_x will be invalid when processing these events. The net result is that the preview window is zoomed x2 (if no model is selected) or the selected model is zoomed x2 and then rotated 57.2958 degrees (1/(2xPi))*360.

If no one can identify WHY these two incorrect events are being generated, my suggestion would be to add state variables for both Zoom and Rotate to only process events between Start=true and End=true inclusive.

Perhaps @cjd or @keithsw1111 can consider this.

**Comment by mmcauley1** [comment]
I am also having the exact same issue described above.  Reproducing the issue and the result are identical.  I am running xlights-2020.20-x86_64.AppImage on Linux Mint 19.3 kernel 5.3.0-51-generic on a Toshiba Satellite laptop.  Using mouse or touchpad makes no difference.

This week I upgraded from 2019.74 which was working correctly.  After experiencing the issue I attempted other 2020.x versions and found the issue in all from 2020.4 on.

leighfiddes you are not alone.
